### ISSUE TITLE
Modal should close when esc key is pressed or user clicks outside

### ISSUE THREAD

**Issue by greatislander** [accessibility-in-action/platform#1067]
**Prerequisites**

- [x] I've [searched](https://github.com/accessibility-exchange/platform/issues) for existing issues

**Describe the bug**

When a modal is visible, the <kbd>esc</kbd> key doesn't close it unless one clicks outside it first. Also, clicking outside should close it.

**To Reproduce**
1. Invoke a modal (e.g. in the translation picker).
2. Press the <kbd>esc</kbd> key.
3. Click outside the modal.
4. Press the <kbd>esc</kbd> key.

**Expected behavior**

- The modal should be closed whenever the <kbd>esc</kbd> key is pressed.
- The modal should be closed when clicking outside.

**Screenshots**

Not applicable.

**Details**
- Device: Mac
- Operating system: macOS 12.5.1
- Browser: Safari 15.6.1

**Additional context**
Not applicable.
